test_conformance/api/test_sub_group_dispatch.cpp

```diff
@@ -75,6 +75,27 @@ int test_sub_group_dispatch(cl_device_id deviceID, cl_context context, cl_comman
     size_t ret_ndrange2d_flattened;
     size_t ret_ndrange3d_flattened;
 
+    if (get_device_cl_version(deviceID) >= Version(3, 0))
+    {
+        int error;
+        cl_uint max_num_sub_groups;
+
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_MAX_NUM_SUB_GROUPS,
+                                sizeof(max_num_sub_groups), &max_num_sub_groups,
+                                NULL);
+        if (error != CL_SUCCESS)
+        {
+            print_error(error, "Unable to get max num subgroups");
+            return error;
+        }
+
+        if (max_num_sub_groups == 0
+            || !is_extension_available(deviceID, "cl_khr_subgroups"))
```

Comment:
I think it is valid for a 3.0 device to report `CL_DEVICE_MAX_NUM_SUB_GROUPS` as non-zero, but to also *not* report the `cl_khr_subgroups` extension because the extension requires IFP whereas the core feature does not.

So the `|| extension` part can be dropped?

If the test actually does require IFP then either a comment or checking `CL_​DEVICE_​SUB_​GROUP_​INDEPENDENT_​FORWARD_​PROGRESS` would be clearer to me.

Reply:
Good spot, I'll update this and check whether the test needs IFP.